### DISCUSSION TITLE
Culling on certain detail meshes is done too early

### DISCUSSION THREAD

**Discussion by LaSaintePapatte** [visualdesigncafe/nature-renderer#52] [Help] [answered]
Hello,

Currently developping a game in a team for our last year in a game design degree, we bought the NR asset which as helped us tremendously for performance issues.
I have just a little issue which I would like to fix : culling on our flowers detail meshes is done too early.

The flowers are made of just a mesh near the petals, that are above the pivot in (0,0,0) which I think is the reason of the early culling.
![image_2024-06-05_170625670](https://github.com/visualdesigncafe/nature-renderer/assets/93184590/8c4d540d-a81f-4e22-b128-6d01fa193de9)

But the problem we encounter is that the NR seems to cull those flowers as soon as their pivot goes off the camera view, which makes them disappear while the petals are still rendered on the screen.
I would like to know if you have any idea as how to fix this, since our deadline is for the end of next week.

https://github.com/visualdesigncafe/nature-renderer/assets/93184590/a1d993cd-0de4-42a1-b817-31e23908ff6a

**Comment by maxplooi** [comment, marked as answer]
Thank you for reporting this. I can confirm that it is a bug in Nature Renderer, the culling algorithm ignores the offset. It should be fixed in the latest version ([2022.0.4](https://visualdesigncafe.com/nature-renderer/docs/2022/release-notes#202204)) which I just published. You can go to **Help > Nature Renderer > Check for Updates** to download the latest version. Please give it a try and let me know if you have any further issues.

**Reply by LaSaintePapatte** [reply]
I updated the package to its latest version and it worked ! Thanks for the responsiveness :)
The flowers don't cull when in the camera anymore. I'll let you know if I have any other issues.

**Reply by westside47** [reply]
@LaSaintePapatte is there any way this will get back ported to 2021? The issue persists there and I really need a fix.

**Reply by westonleavitt** [reply]
@maxplooi is there any way this will get back ported to the 2021 version? This issue exists there as well and is very frustrating

**Comment by stonstad** [comment]
Related: https://github.com/visualdesigncafe/nature-renderer/issues/80

**Comment by westonleavitt** [comment]
@maxplooi

**Comment by westonleavitt** [comment]
I fixed this on the compute shader, there is a miscalculation. Reach out for fix
